### ISSUE TITLE
How to make interaction between electron-chrome-extension and background scripts of chrome extension?

### ISSUE THREAD

**Issue by Jenifer-TheCoder** [samuelmaddock/electron-browser-shell#8]
I have a feeling that am missing something from the document of electron-browser-shell. Here are the steps i've added:

1. Loaded chrome extension by placing unpacked extension folder under Extensions directory. Loading works and that was obvious with extension icon on left side of Electron app toolbar
2. Wherever i have chrome.tabs*,chrome.windows* API calls from background scripts of extension, I get console messages from saying **undefined**.
3. Am i missing anything?
4. As per readme document, it says I had to create extensions instance and addtab.I hope it has to be inside main-process which is already in place.(Since i just use electron-browser-shell as an Electron application)

**Comment by samuelmaddock** [comment]
What you've described sounds correct. Without code, I won't be able to reproduce the problems you're seeing. Please consider creating a minimal example extension with the same issue.

**Comment by Jenifer-TheCoder** [comment]
Ok.
Front-end requirements:

- Get electron-browser-shell extension code and launch using "yarn start" command
Backend requirements:

- Place the attached extension folder into extensions directory
[ioknjhjjcencfpdllmeoijbacekgbgdg.zip](https://github.com/samuelmaddock/electron-browser-shell/files/5867268/ioknjhjjcencfpdllmeoijbacekgbgdg.zip)
-For some reason background scripts weren't visible in content-scripts view, hence I had to add content-scripts section into manifest.json
-I had added sample snippet to get active tab in [extensions\ioknjhjjcencfpdllmeoijbacekgbgdg\1.0.3.19_0\grabber\assima.grabber.background.tabs.js]
- Which will fail to retrieve current tab saying chrome.tabs as undefined, this is the problem I am facing with few other APIs as well(Like chrome.tabs.sendMessage)

Any suggestions would be appreciated, thanks in advance @samuelmaddock

**Comment by samuelmaddock** [comment]
I seemed to be able to run the extension and receive the result of the active tab.

![image](https://user-images.githubusercontent.com/1656324/106348790-eca84800-6296-11eb-8788-6cdaed11dc60.png)

Try getting the latest changes from this repository and run `yarn start:debug`. It should open DevTools for the background script (like what I have in the screenshot). There you can access APIs such as `chrome.tabs.*`, `chrome.windows.*`, etc.

**Comment by Jenifer-TheCoder** [comment]
Thanks for your efforts @samuelmaddock , highly appreciate.

- I had checked with yarn start:debug flag, could see chrome.Tabs getting invoked inside electron-chrome-extensions.
- I am with new bottleneck with chrome.webNavigation.getAllFrames API, where it doesn't return frames of tab.
- In the shared extension folder, there is a function loadFrameHierarchy inside assima.grabber.background.tabs.js file.
- Since frame array=>arr is of zero count,extension doesn't work as expected
- Hopefully chrome.webNavigation.getAllFrames is supported in electron-chrome-extensions.
Any idea why it doesn't return frames of the tab?

**Comment by samuelmaddock** [comment]
It's a bit hidden away, but under the [list of supported APIs](https://github.com/samuelmaddock/electron-browser-shell/tree/master/packages/electron-chrome-extensions#supported-chrome-apis) I mention that `getFrame` and `getAllFrames` require an upcoming feature in Electron 12. You can try it out by upgrading to the beta.
```
cd packages/shell
yarn add electron@12.0.0-beta.20
cd ../..
yarn start:debug
```

**Comment by Jenifer-TheCoder** [comment]
@samuelmaddock ,thanks for your life-saving support :-)

Now have to check how this can be wrapped as an extension for Electron applications with an unpacked version.

**Comment by samuelmaddock** [comment]
The error you're seeing should be fixed within a couple weeks and before Election 12 is stable. You can follow electron/electron#26824 for progress.

I'm going to close this issue now. For general questions, you can create a thread in Discussions and if you find other problems you can raise an issue.

**Comment by samuelmaddock** [comment]
@Jenifer-TheCoder fixed in `12.0.0-beta.24`
